app/models/target.rb

```diff
@@ -27,12 +29,19 @@ class Target < ActiveRecord::Base
     food: 8
   }
 
+  belongs_to :user
+
   validates :area_lenght, :title, presence: true
   validates :topic, presence: true, inclusion: { in: topics.keys }
   validates :latitude, presence: true,
                        numericality: { greater_than_or_equal_to: -90, less_than_or_equal_to: 90 }
   validates :longitude, presence: true,
                         numericality: { greater_than_or_equal_to: -180, less_than_or_equal_to: 180 }
+  validate :validate_target_limit
```

Comment:
By default, this will run every time you save this object. As you only need to validate this on creation you could add `on: :create`.
You can find more information about custom validations in [here](https://guides.rubyonrails.org/active_record_validations.html#custom-methods)